src/Illuminate/Support/helpers.php

```diff
@@ -196,6 +197,54 @@ function literal(...$arguments)
     }
 }
 
+if (! function_exists('lazy') && version_compare(phpversion(), '8.4.0', '>=')) {
+    /**
+     * Create a lazy instance.
+     *
+     * @template TValue of object
+     *
+     * @param  class-string<TValue>|(\Closure(TValue): mixed)  $class
+     * @param  (\Closure(TValue): mixed)|int  $callback
+     * @param  int  $options
+     * @param  array<string, mixed>  $eager
+     * @return TValue
+     */
+    function lazy($class, $callback = 0, $options = 0, $eager = [])
```

Comment:
What if both `$callback` and `$options` are `int`?
```php
lazy(MyClass::class); // $callback and $options both default to 0
lazy(MyClass::class, options: 42); // It will fail, because 0 cannot be called and newLazyGhost() does not accept 42 (only 0 and 8)
```

Reply:
Yeah, but it would be better if it was _explicitly_ validated so there could be a _descriptive_ exception message. This can happen easily as the method signature doesn't clearly describe the options without consulting the docs.